frontend/pages/admin/TeamManagementPage/TeamDetailsWrapper/TeamDetailsWrapper.tsx

```diff
@@ -308,15 +313,31 @@ const TeamDetailsWrapper = ({
       dispatch(teamActions.update(currentTeam?.id, updatedAttrs))
         .then(() => {
           dispatch(teamActions.loadAll({ perPage: 500 }));
-          dispatch(renderFlash("success", "Team updated"));
+          dispatch(
+            renderFlash(
+              "success",
+              `Successfully updated team name to ${updatedAttrs?.name}`
+            )
+          );
+          setBackendValidators({});
           refetchTeams();
           refetchMe();
-          // TODO: error handling
+          toggleEditTeamModal();
         })
-        .catch(() => null);
-      toggleEditTeamModal();
+        .catch((updateError: IOldApiError) => {
+          if (updateError.base.includes("Duplicate")) {
```

Comment:
Instead of parsing the response object for a string, could we check the response status code? It's currently returning 409 if a team name already exists, which is the status code for "conflict". 

The benefit of checking status codes is that they are unlikely to change, whereas any change to the response object could break the string parsing.